embassy-rp/src/clocks.rs

```diff
@@ -189,14 +629,56 @@ unsafe fn configure_pll(p: pac::pll::Pll, refdiv: u32, vco_freq: u32, post_div1:
 
     // Wait for PLL to lock
     p.prim().write(|w| {
-        w.set_postdiv1(post_div1);
-        w.set_postdiv2(post_div2);
+        w.set_postdiv1(config.post_div1);
+        w.set_postdiv2(config.post_div2);
     });
 
     // Turn on post divider
     p.pwr().modify(|w| w.set_postdivpd(false));
 }
 
+pub struct Gpout0 {
+    _pin: crate::peripherals::PIN_21,
+}
+
+impl Gpout0 {
+    pub fn new(pin: crate::peripherals::PIN_21) -> Self {
```

Comment:
this should have a `trait GpoutPin`, and take a `pin: impl Peripheral<P = impl GpoutPin> + 'd`. 

This way the same struct can be used for all 4 GPOUTs, and can be created with a borrowed pin so you can drop it and get the pin back Same as with all other drivers that use pins.

Reply:
I'm going to need some help with this. I think I addressed the drop part, but I'm not sure how to handle the 4 pins with out a huge mess of match statements to route the commands to the right registers and such.

Reply:
oh I see, the `GPOUTx` regs are duplicated in `rp-pac`. 

You can try adding some transforms [here](https://github.com/embassy-rs/rp-pac/blob/main/svd/rp2040.yaml) (`MergeEnums`, `MergeFieldsets`, `MakeRegisterArray` should do the trick), then run `update.sh`.

Once the regs are arrayified, it should be easy to do. For example, add a `fn number() -> u8` to `GpoutPin`, store the num in `self`, so you can use it as an array index.

Reply:
Fixed up, but should certainly be re-reviewed.

Reply:
:+1: :ok_hand: